src/components/EditProfile/index.jsx

```diff
@@ -78,7 +80,8 @@ const EditProfile = ({ userData, username, setIsEditing, setUserData }) => {
     if (!usernameAvailable) {
       return;
     }
-    if (image) {
+    const isImgDeleted = deleteImg(userData.avatar)
```

Comment:
isn't that an async call and should be awaited? Also, we should be deleting the image only after new one replaces it. you may keep this url in variable and call it after update is successful to free the resources